### DISCUSSION TITLE
Sequential filling up of disks and drive spindown

### DISCUSSION THREAD

**Discussion by jpmvferreira** [trapexit/mergerfs#1245] [Q&A] [answered]
I've got a server with both SSDs and HDDs, and I want them all under one folder, with one the data moving from one disk to the other only after the previous one is full. As far as I know, this is MergerFS's default behavior.

This sequential writing is important because I want to keep the drives idle as much as possible.
After reading the wiki page on "Limit Drive Spinup" and a similar discussion in this repository a while ago, I have a few questions and I want to ensure that I understood it correctly.
If I have an array of disks in a MergerFS pool and request a file, which can only be on one disk, are all disks required to spin up to find it? 
And, once that file is found on a given disk, will the other disks be able to spin down?

Also, unrelated to MergerFS, is there a difference between the power consumption an SSD in the idle state and in the woken-up-but-not-doing-any-operations state (if there is such a state)?

**Comment by trapexit** [comment, marked as answer]
> I've got a server with both SSDs and HDDs, and I want them all under one folder, with one the data moving from one disk to the other only after the previous one is full. As far as I know, this is MergerFS's default behavior.

mergerfs does no automatic moving of files. It does placement at create time.

>This sequential writing is important because I want to keep the drives idle as much as possible.

You can't do that that way. https://github.com/trapexit/mergerfs/wiki/Limit-Drive-Spinup

> If I have an array of disks in a MergerFS pool and request a file, which can only be on one disk, are all disks required to spin up to find it?

Yes. How else would it know where the file is located? Something must at some point find where the file is located and keep track of it. Users can completely change things out of band and there is no real way to know that without looking. Looking means spinup most of the time. Any interaction with the drive that isn't cached by the OS leads to spinup.

> And, once that file is found on a given disk, will the other disks be able to spin down?

It depends on the state of the system. If the drive is in use, no. If it isn't, yes.

> Also, unrelated to MergerFS, is there a difference between the power consumption an SSD in the idle state and in the woken-up-but-not-doing-any-operations state (if there is such a state)?

It will depend on the device. You have to read about the device in question. If the ssd has low power modes then it will use less power when idle. Every ssd is different is design, features, etc.

**Comment by xsm1997** [comment]
In my practice, I have a 2\*16T and 2\*4T HDD mergerfs pool in my NAS. Most of my files are in one of the 16T HDD. I can tell that with a decent Linux setup and enough RAM (e.g., 32GB for me), the filesystem cache of Linux should not make the other 3 drives spinned up frequently, at least in my case.

**Reply by jpmvferreira** [reply]
Okay that's good to know, although I will be using a low-end system with 2 SSD's (1TB + 500GB), 3 HDD's (1 TB + 500 GB + 300 GB) and 8 GB of RAM.
I know that it is possible to have the metadata separate from the contents of the files themselves, which would solve the problem, but as far as I know only ZFS does that, not even Btrfs, which as far as I know would require a beefier machine than mine, which is made for low power usage and high idle states.
